EventFilter/HcalRawToDigi/plugins/HcalDigiToRawuHTR.cc

```diff
@@ -0,0 +1,297 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+
+#include "EventFilter/HcalRawToDigi/interface/HcalUHTRData.h"
+#include "DataFormats/HcalDigi/interface/HcalQIESample.h"
+
+#include "CondFormats/HcalObjects/interface/HcalElectronicsMap.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/FEDRawData/interface/FEDRawDataCollection.h"
+#include "DataFormats/FEDRawData/interface/FEDHeader.h"
+#include "DataFormats/FEDRawData/interface/FEDTrailer.h"
+#include "DataFormats/FEDRawData/interface/FEDNumbering.h"
+#include "DataFormats/FEDRawData/interface/FEDRawData.h"
+#include "FWCore/Utilities/interface/CRC16.h"
+
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+#include "CalibFormats/HcalObjects/interface/HcalDbService.h"
+#include "CalibFormats/HcalObjects/interface/HcalDbRecord.h"
+
+#include "PackerHelp.h"
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+
+/* QUESTION: what about dual FED readout? */
+/* QUESTION: what do I do if the number of 16-bit words
+   are not divisible by 4? -- these need to
+   fit into the 64-bit words of the FEDRawDataFormat */
+
+
+using namespace std;
+
+class HcalDigiToRawuHTR : public edm::EDProducer {
```

Comment:
this should be of a type known to threaded framework
https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkModuleTypes
edm::global would be a better choice if implementation allows, edm::stream is OK.
edm::one is rather an exception and will need some justification.